### PR TITLE
Fix some macOS dependencies

### PR DESCRIPTION
**Pull request by kneep**
Hi all. It seems building on and for macOS is not officially supported in the latest release. Some packages are missing for ```osx```, while a few others are special because they are already shipped with macOS or Xcode. At least these changes need to be made to make ```rosdep``` happy.

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -546,9 +546,6 @@ clang-tidy:
   fedora: [clang-tools-extra]
   gentoo: [sys-devel/clang]
   nixos: [clang]
-  osx:
-    homebrew:
-      packages: [llvm]
```
Comment (ivanpauno) [inline rosdep/base.yaml:551]: why is this deleted?

IIUC, llvm includes clang-tidy when installed

File: rosdep/base.yaml [hunk 2]
```diff
@@ -5608,9 +5608,6 @@ libxml2-utils:
   gentoo: [dev-libs/libxml2]
   nixos: [libxml2]
   openembedded: [libxml2@openembedded-core]
-  osx:
-    homebrew:
-      packages: [libxml2]
```
Comment (ivanpauno) [inline rosdep/base.yaml:5613]: why is this getting deleted?

File: rosdep/osx-homebrew.yaml [hunk 3]
```diff
@@ -154,7 +158,7 @@ graphviz:
 gtest:
   osx:
     homebrew:
-      packages: [gtest]
+      packages: [googletest]
```
Comment (ivanpauno) [inline rosdep/osx-homebrew.yaml:161]: why is this repeated in base and here? I'm not sure how that works

Comment (kneep) [reply rosdep/osx-homebrew.yaml:161]: @ivanpauno  I have the same doubt. I see other macOS packages randomly scattering in these two files. Maybe someone should clean all these up.

File: rosdep/base.yaml (hunk 1 above)
Comment (ivanpauno) [reply rosdep/base.yaml:551]: > So to make clang-tidy work, we need to manually put it to $PATH.

Which one? The one provided by Xcode?
If that's the case, removing the rule sounds fine, but if installing `llvm` through homebrew is still needed I wouldn't delete the rule.

Comment (kneep) [reply rosdep/base.yaml:551]: @ivanpauno I mean the homebrew one. Xcode does not include a ```tidy-clang```. But even after installing ```llvm``` from homebrew, ```rosdep``` still fails because it query the installation with 
```
brew info llvm --json=v1
```
and the check the ```linked_keg``` in the output. For ```keg-only``` packages, ```linked_keg``` key is null so ```rosdep``` fails. You can take a look at this code:
https://github.com/ros-infrastructure/rosdep/blob/9f5829869db6c28d3cb0a536c699b82919a4c358/src/rosdep2/platforms/osx.py#L207

It seems currently there is no easy way to automatically install ```clang-tidy```. As long as we install it with homebrew (either by ```rosdep``` or manually), we have to manually link the executable to ```$PATH```. But removing ```llvm``` from ```base.yaml``` will at least make ```rosdep``` happy and maybe instructions should be added to documentations.

Comment (ivanpauno) [reply rosdep/base.yaml:551]: that sounds fair enough to me

**Review by ivanpauno — APPROVED** [review]

**Review by ivanpauno — APPROVED** [review]

File: rosdep/python.yaml [hunk 4]
```diff
@@ -6904,6 +6910,9 @@ python3-mypy:
   nixos: [python3Packages.mypy]
   openembedded: [python3-mypy@meta-ros-common]
   opensuse: [mypy]
+  osx:
+    pip:
```
[the remark this answers is not in the document]
Comment (sloretz) [reply rosdep/python.yaml:6914]: I don't see any guidelines on which is preferred, so I'd say this is fine as is.

File: rosdep/python.yaml [hunk 5]
```diff
@@ -8270,6 +8291,9 @@ python3-venv:
   fedora: [python3-libs]
   nixos: [python3]
   opensuse: [python3-virtualenv]
+  osx:
+    pip:
+      packages: [virtualenv]
```
[the remark this answers is not in the document]
Comment (sloretz) [reply rosdep/python.yaml:8296]: > So leave this entry blank?

Leaving blank sounds right to me!

**Comment by tfoote** [comment]
@kneep Do you have a moment to resolve the final comments on this?

File: rosdep/osx-homebrew.yaml [hunk 6]
```diff
@@ -583,6 +595,10 @@ python-vtk:
   osx:
     homebrew:
       depends: [libvtk]
+python3:
+  osx:
+    homebrew:
+      packages: []
```
[the remark this answers is not in the document]
Comment (kneep) [reply rosdep/osx-homebrew.yaml:601]: I'll remove this patch in the push. I might get this wrong.

**Comment by kneep** [comment]
@tfoote Hi Tully. I just resolved all the conversations and pushed revised patches, removing the python3 patch and leaving the virtualenv blank according to the discussion with @sloretz. Thanks.

**Review by clalancette — APPROVED** [review]
